include/sys/trace.h

```diff
@@ -43,19 +43,23 @@
  */
 #undef DTRACE_PROBE1
 #define	DTRACE_PROBE1(name, t1, arg1) \
-	trace_zfs_##name((arg1))
+	if (arg1) \
+		trace_zfs_##name((arg1))
 
 #undef DTRACE_PROBE2
 #define	DTRACE_PROBE2(name, t1, arg1, t2, arg2) \
-	trace_zfs_##name((arg1), (arg2))
+	if (arg1 && arg2) \
```

Comment:
> Are the NULL entries manually updated to denote the expected results?

Exactly, it was just to show what i expected from this change.

I'm not sure if the code was originally implemented this way so we could trace a NULL dbuf (which seems would only happen when the dbuf cache is already empty), other platforms share this same code so i doubt it was done this way by chance.

Someone with better understanding of dbuf eviction would probably know, my concern is to minimize code differences with other implementations.